src/Mod/Part/App/AttachExtension.cpp

```diff
@@ -607,6 +607,31 @@ void AttachExtension::handleLegacyTangentPlaneOrientation()
         }
 
         // convert placement and expressions according to the dominant axis
+        auto makeRotatedExpression =
+            [owner](const App::Expression* expr, double angle) -> App::Expression* {
+            if (!expr) {
+                return nullptr;
+            }
+
+            std::string s = "(" + expr->toString() + ")";
+            if (angle >= 0) {
+                s += " + " + std::to_string(angle);
+            }
+            else {
+                s += " - " + std::to_string(-angle);
+            }
+
+            if (const App::Expression* simple = expr->eval()) {
+                if (auto ue = dynamic_cast<const App::UnitExpression*>(simple)) {
```

Comment:
In cases where you're casting FreeCAD's types, use `freecad_cast` instead of `dynamic_cast`.

Reply:
Odd - is the type information wrong for them? They are derived from `Base::BaseClass`, so I think `freecad_cast` should work.